src/cljs/athens/hotkeys.cljs

```diff
@@ -0,0 +1,75 @@
+(ns athens.hotkeys
+  (:require
+    ["react-hotkeys" :refer [GlobalHotKeys HotKeys configure]]
+    [re-frame.core :refer [dispatch subscribe]]
+    [athens.util :as util]
+    [athens.router :as router]))
+
+(configure #js{:ignoreTags #js[]})
+
+(def modkey
+  (let [os (util/get-os)]
+    (or (and (= os :mac) "meta")
+      (and (= os :windows) "ctrl")
+      (and (= os :linux) "ctrl"))))
+
+(defn with-mod
+  [combination]
+  (str modkey "+" combination))
+
+(defn not-editing
+  []
+  (nil? @(subscribe [:editing/uid])))
+
+(defn selected-items-not-empty
+  []
+  (not-empty @(subscribe [:selected/items])))
+
+(defn maybe-undo-or-redo
+  [operation]
+  (when
+    (or (not-editing)
+      (selected-items-not-empty))
+    (dispatch [operation])))
+
+(defn global-hotkeys
+  []
```

Comment:
Bind reactive values here in `let`.
Like `not-editing`.
And return rendering fn.

Reply:
That works, but there is a bit of a possible performance overhead of it. This component is rendered at the root level of the app, so each time `not-editing` changes (not super often), it will rerender. 

Having the subscription (not actually a subscription, values is just "read" from the re-frame store) inside the event handler, avoids those rerenders.

Anothe possible overhead is that, if we subscribe here, the event handlers passed to GlobalHotKeys won't have an stable identity (function changes on each render), with may also have another overhead with the listening/unlistening of the DOM (this worries me less than the one I mentioned before)

Reply:
If we can't avoid re-rendering, then subscription in event handler is 👍 .